src/WingetCreateTests/WingetCreateTests/UnitTests/UpdateCommandTests.cs

```diff
@@ -185,8 +186,26 @@ public async Task UpdateFailsWithUnmatchedPackages()
             var updatedManifests = await command.DeserializeExistingManifestsAndUpdate(initialManifestContent);
             Assert.IsNull(updatedManifests, "Command should have failed");
             string result = this.sw.ToString();
-            Assert.That(result, Does.Contain(Resources.MultipleInstallerUpdateDiscrepancy_Error), "Installer discrepency error should be thrown");
-            Assert.That(result, Does.Contain(string.Format(Resources.MissingPackageError_Message, InstallerType.Msix, InstallerArchitecture.X86)), "Missing package error should be thrown");
+            Assert.That(result, Does.Contain(Resources.NewInstallerUrlMustMatchExisting_Message), "Installer must have match error should be thrown");
+        }
+
+        /// <summary>
+        /// Since some installers are incorrectly labeled on the manifest, resort to using the installer URL to find matches.
+        /// This unit test uses a msi installer that is not an arm64 installer, but because the installer URL includes "arm64", it should find a match.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Test]
+        public async Task UpdateBasedOnInstallerUrlMatch()
+        {
+            TestUtils.TestInstallerBaseUrl = "https://fakedomain.com/arm64/";
+            TestUtils.InitializeMockDownloads(TestConstants.TestMsiInstaller);

```

Comment:
You can pass in multiple URLs by changing InitializeMockdownloads

Reply:
Reverted to what we had before. I decided to go with your suggestion about using Path.GetFileName() to separate out the architecture from the filename.